pkg/rc/manager.go

```diff
@@ -0,0 +1,183 @@
+package rc
+
+import (
+	"github.com/square/p2/Godeps/_workspace/src/github.com/Sirupsen/logrus"
+
+	"github.com/square/p2/pkg/kp/rcstore"
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/rc/fields"
+)
+
+// The RCManager is responsible for spawning and reaping replication controllers
+// as they are added to and deleted from Consul. Multiple managers can exist
+// simultaneously, but each one must hold a different Consul session. This
+// ensures that the managers do not instantiate the same replication controller
+// multiple times.
+type RCManager struct {
```

Comment:
Personal beef: I hate the word "manager" and would prefer to limit its use as much as possible. The only reason I was okay with "SessionManager" is because it has precedent in other frameworks. I prefer your term "RCFarm" honestly.